buildSrc/src/main/kotlin/Toolchain.kt

```diff
@@ -51,4 +52,17 @@ private fun KotlinCompile.setupKotlinToolchain(javaToolchains: JavaToolchainServ
     kotlinJavaToolchain.toolchain.use(javaToolchains.launcherFor {
         languageVersion.set(JavaLanguageVersion.of(jdkToolchainVersion))
     })
+}
+
+fun KotlinCompile.getAccessToInternalDefinitionsOf(vararg projects: Project) {
+    projects.forEach { project ->
+        val mainSourceSet = project.extensions
+            .getByType(JavaPluginExtension::class.java).sourceSets
+            .getByName("main").output.classesDirs
+        val jarArchive = Paths.get(project.buildDir.absolutePath, "libs",
```

Comment:
Please fix deprecated `buildDir`. There should already be some examples in the main `build.gradle.kts` script.